targets/DeployDependencies.proj

```diff
@@ -129,6 +129,12 @@
           DestinationFolder="@(RuntimeProjectJson->'%(TestDeploymentDir)')"
           SkipUnchangedFiles="true"
           />
+
+    <Copy SourceFiles="@(ResolvedRuntimeFiles)"
```

Comment:
Does this mean we can just get rid of `DeploymentDir` in favor of `OutputPath`? AFAIK, initially, we wanted  `OutputPath` to only contain compilation output, `DeploymentDir` to contain `OutputPath` + dependencies + runtime, and `TestDeploymentDir` to contain `DeploymentDir` + test dependencies. I think the original idea was to do one .net core build which would produce one `OutputPath` (anycpu, contents would be the same) but different deployment dirs for each OS (win, linux, osx).

Since we never ended up doing the multi deployment, I'd be in favor of this PR also getting rid of [`DeploymentDir`](https://github.com/Microsoft/msbuild/blob/master/dir.props#L409) to eliminate duplicate crud. And then re-orient the [`RuntimeProjectJson`](https://github.com/Microsoft/msbuild/blob/6851538897f5d7b08024a6d8435bc44be5869e53/targets/DeployDependencies.proj#L90) to `OutputPath`.

A good test for this change would be the Microbuild build